compiler-k2/src/main/kotlin/com/squareup/anvil/compiler/k2/fir/contributions/BindingModuleData.kt

```diff
@@ -0,0 +1,85 @@
+package com.squareup.anvil.compiler.k2.fir.contributions
+
+import com.squareup.anvil.compiler.k2.util.AnvilNames
+import com.squareup.anvil.compiler.k2.util.DaggerNames
+import com.squareup.anvil.compiler.k2.util.classId
+import com.squareup.anvil.compiler.k2.util.toFirAnnotation
+import org.jetbrains.kotlin.descriptors.ClassKind
+import org.jetbrains.kotlin.fir.FirSession
+import org.jetbrains.kotlin.fir.declarations.getKClassArgument
+import org.jetbrains.kotlin.fir.expressions.FirAnnotation
+import org.jetbrains.kotlin.fir.expressions.FirAnnotationCall
+import org.jetbrains.kotlin.fir.expressions.FirNamedArgumentExpression
+import org.jetbrains.kotlin.fir.expressions.builder.buildAnnotation
+import org.jetbrains.kotlin.fir.expressions.impl.FirAnnotationArgumentMappingImpl
+import org.jetbrains.kotlin.fir.extensions.ExperimentalTopLevelDeclarationsGenerationApi
+import org.jetbrains.kotlin.fir.extensions.FirExtension
+import org.jetbrains.kotlin.fir.plugin.createTopLevelClass
+import org.jetbrains.kotlin.fir.resolve.fqName
+import org.jetbrains.kotlin.fir.symbols.impl.FirClassLikeSymbol
+import org.jetbrains.kotlin.fir.symbols.impl.FirClassSymbol
+import org.jetbrains.kotlin.fir.types.ConeKotlinType
+import org.jetbrains.kotlin.fir.types.builder.buildResolvedTypeRef
+import org.jetbrains.kotlin.fir.types.classId
+import org.jetbrains.kotlin.fir.types.constructClassLikeType
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlin.name.Name
+
+@OptIn(ExperimentalTopLevelDeclarationsGenerationApi::class)
+public class BindingModuleData(
```

Comment:
I kept the structure here mostly in-line with the POC branch until we can discuss concerns/alternatives (related https://github.com/square/anvil/pull/1110/files#r1951710072), and also because I'd like us to get the merging impl + related integration tests in before doing any heavier refactoring.